File: ip-address-validation-nodejs-examples/get_geo_location_by_ipv4_soap_sdk_example.js
import { GetGeoLocationByIPV4Soap } from "../ip-address-validation-nodejs/SOAP/get_geo_location_by_ipv4_soap.js";

export async function getGeoLocationByIPV4SoapGO(licenseKey, isLive) {
  console.log("\n--------------------------------------------------------");
  console.log("IP Address Validation - GetGeoLocationByIP_V4 - SOAP SDK");
  console.log("--------------------------------------------------------");

  const ipAddress = "72.205.70.231";
  const timeoutSeconds = 15;

  console.log("\n* Input *\n");
  console.log(`IP Address     : ${ipAddress}`);
  console.log(`License Key    : ${licenseKey}`);
  console.log(`Is Live        : ${isLive}`);
  console.log(`Timeout Seconds: ${timeoutSeconds}`);

  try {
    const ipav4 = new GetGeoLocationByIPV4Soap(ipAddress, licenseKey, isLive, timeoutSeconds);
    const response = await ipav4.getGeoLocationByIPV4();

    if (response.Error) {
        console.log("\n* Error *\n");
        console.log(`Error Desc    : ${response.Error.Desc}`);
        console.log(`Error Number  : ${response.Error.Number}`);
        console.log(`Error Location: ${response.Error.Location}`);
        return;
    }

    console.log("\n* GeoLocation Info *\n");
    if (response) {
      console.log(`Certainty      : ${response.Certainty}`);
      console.log(`City           : ${response.City}`);
      console.log(`Region         : ${response.Region}`);
      console.log(`Country        : ${response.Country}`);
      console.log(`Country ISO3   : ${response.CountryISO3}`);
      console.log(`Country ISO2   : ${response.CountryISO2}`);
      console.log(`Postal Code    : ${response.PostalCode}`);
      console.log(`Metro Code     : ${response.MetroCode}`);
      console.log(`DMA            : ${response.DMA}`);
      console.log(`State FIPS     : ${response.StateFIPS}`);
      console.log(`County FIPS    : ${response.CountyFIPS}`);
      console.log(`Latitude       : ${response.Latitude}`);
      console.log(`Longitude      : ${response.Longitude}`);
      console.log(`Is Proxy       : ${response.IsProxy}`);
      console.log(`Proxy Type     : ${response.ProxyType}`);
      console.log(`Possible Mobile: ${response.PossibleMobileDevice}`);
      console.log(`ISP            : ${response.ISP}`);
      console.log(`Netblock Owner : ${response.NetblockOwner}`);
      console.log(`Host Names     : ${response.HostNames}`);
      console.log(`IP Note Codes  : ${response.IPNoteCodes}`);
      console.log(`IP Notes       : ${response.IPNotes}`);
      console.log(`Debug          : ${response.Debug}`);
    } else {
      console.log("No geolocation info found.");
    }

  
  } catch (e) {
    console.log(`\nException occurred: ${e.message}`);
  }
}

File: ip-address-validation-python-examples/get_geo_location_by_ipv4_soap_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../ip-address-validation-python/SOAP"))

from get_geo_location_by_ipv4_soap import GetGeoLocationByIPV4Soap

def get_geo_location_by_ipv4_soap_sdk_go(is_live: bool, license_key: str) -> None:
    print("\n--------------------------------------------------------")
    print("IP Address Validation - GetGeoLocationByIP_V4 - SOAP SDK")
    print("--------------------------------------------------------")

    ip_address = "72.205.70.231"
    timeout_seconds = 15

    print("\n* Input *\n")
    print(f"IP Address     : {ip_address}")
    print(f"License Key    : {license_key}")
    print(f"Is Live        : {is_live}")
    print(f"Timeout Seconds: {timeout_seconds}")

    try:
        service = GetGeoLocationByIPV4Soap(
            license_key,
            is_live,
            timeout_ms=timeout_seconds * 1000
        )
        response = service.get_geo_location_by_ip_v4(ip_address)

        if not hasattr(response, "Error"):
            print("\n* GeoLocation Info *\n")
            if response:
                print(f"Certainty      : {response.Certainty}")
                print(f"City           : {response.City}")
                print(f"Region         : {response.Region}")
                print(f"Country        : {response.Country}")
                print(f"Country ISO3   : {response.CountryISO3}")
                print(f"Country ISO2   : {response.CountryISO2}")
                print(f"Postal Code    : {response.PostalCode}")
                print(f"Metro Code     : {response.MetroCode}")
                print(f"DMA            : {response.DMA}")
                print(f"State FIPS     : {response.StateFIPS}")
                print(f"County FIPS    : {response.CountyFIPS}")
                print(f"Latitude       : {response.Latitude}")
                print(f"Longitude      : {response.Longitude}")
                print(f"Is Proxy       : {response.IsProxy}")
                print(f"Proxy Type     : {response.ProxyType}")
                print(f"Possible Mobile: {response.PossibleMobileDevice}")
                print(f"ISP            : {response.ISP}")
                print(f"Netblock Owner : {response.NetblockOwner}")
                print(f"Host Names     : {response.HostNames}")
                print(f"IP Note Codes  : {response.IPNoteCodes}")
                print(f"IP Notes       : {response.IPNotes}")
                print(f"Debug          : {response.Debug}")
            else:
                print("No geolocation info found.")
        else:
            if hasattr(response, "Error") and response.Error:
             print("\n* Error *\n")
             print(f"Error Desc    : {response.Error.Desc}")
             print(f"Error Number  : {response.Error.Number}")
             print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print(f"\nException occurred: {str(e)}")